### ISSUE TITLE
CachedExternal breaks when using URL onlineResource.

### ISSUE THREAD

**Issue by fellahst** [Geomatys/geotoolkit#4]
The CachedExternal (referring to an ExternalGraphic) breaks when the online resource points  to a resource on the web using http protocol.
The   Path path = Paths.get(uri) will break on the http scheme (No FileSystemProvider exists for http scheme). Version 4.0-M4 seems to work.

Here the snippet of code that causes problem.

 //if no inline image then use the online image
        if(cachedImage == null){
            OnlineResource online = styleElement.getOnlineResource();
            if(online != null && online.getLinkage() != null){

```
            final URI uri = styleElement.getOnlineResource().getLinkage();
            isSVG = uri != null && uri.toString().toLowerCase().endsWith(".svg");

            if (!isSVG) {
                **final Path path = Paths.get(uri);**
                try (InputStream in = IOUtilities.open(path)){
                    cachedImage = ImageIO.read(in);
                } catch (IOException ex) {
                    LOGGER.log(Level.INFO, "CacheExternal, can read image for path : "+ uri.toString()+" : "+ex.getMessage());
                }
            }
        }
    }
```

**Comment by desruisseaux** [comment]
A fix has been pushed on master. Unfortunately this fix is mixed with an ongoing work on replacing JSR-275 by JSR-363 (I should have done that work on a separated branch - sorry for that), so doing a checkout now may cause a temporary regression in a few operations related to units of measurement. I will post a new comment here after the migration has been completed.

**Comment by desruisseaux** [comment]
It should be fixed in release 4.0.0 of Geotk. Note that this is the first "release" after a series of milestones (the last one was 4.0-M5), but I see this 4.0.0 release a little bit as a milestone too. The reason is that it is the last version with the old Feature model inherited from GeoTools days. Starting with 5.0-SNAPSHOT, we use a new Feature model implemented in Apache SIS. This change has a wide impact, so we may have to publish some other updates in the 4.x branches in the future.
